File: examples_ios/MyApp/Sources/Domain/TriggerManager.swift
import ConfirmitMobileSDK
import Foundation

class TriggerManager {
  func getPrograms() -> [Program] {
    var programs: [Program] = []
    let servers = try! ConfirmitServer.getServers()
    for server in servers {
      do {
        let program = try TriggerSDK.getPrograms(serverId: server.serverId)
        programs += program
      } catch {
        // server doesn't exist
      }
    }
    return programs.sorted(by: { (p1, p2) -> Bool in
      p1.serverId > p2.serverId
        })
  }

  func setAllDelegate() {
    for program in getPrograms() {
      setDelegate(serverId: program.serverId, programKey: program.programKey)
    }
  }

  func setDelegate(serverId: String, programKey: String) {
    TriggerSDK.setCallback(serverId: serverId, programKey: programKey, callback: TriggerCallback())
  }
}

File: examples_android/app/src/main/java/com/example/myapp/domain/TriggerManager.kt
package com.example.myapp.domain

import com.confirmit.mobilesdk.ConfirmitServer
import com.confirmit.mobilesdk.TriggerSDK
import com.confirmit.mobilesdk.database.externals.Program

class TriggerManager {
    fun getPrograms(): List<Program> {
        val programs: MutableList<Program> = mutableListOf()
        val servers = ConfirmitServer.getServers()
        for (server in servers) {
            try {
                val program = TriggerSDK.getPrograms(server.serverId)
                programs.addAll(program)
            } catch (_: Exception) {
            }
        }

        return programs.sortedByDescending { it.serverId }
    }

    fun setAllDelegate() {
        for (program in getPrograms()) {
            setDelegate(program.serverId, program.programKey)
        }
    }

    fun setDelegate(serverId: String, programKey: String) {
        TriggerSDK.setCallback(serverId, programKey, TriggerCallback())
    }
}